fgsm_calculations.py

```diff
@@ -3,115 +3,138 @@
-    feature_num: Integer for feature for which to calculate gradient with respect to. Should be a number between 20-39, corresponding to embedding dimension.
-
-  Returns:
-    Gradient of loss with respect to singular feature embedding.
-  """
-  features = model_builder.normalized_features_evaluated
-  direction = 0
-  layer_size = int(constants._HIDDEN_LAYER_DIMS[0]) 
-  for num in range(layer_size):
-    grad = model_builder.grad_variable_pair_evaluated[0][feature_num][num] 
-    weight = model_builder.grad_variable_pair_evaluated[1][feature_num][num]
-    feature = features[answer_num][feature_num] 
-    if feature != 0:
-      direction += grad * (weight/feature) #mathmatically feature value has no impact here since in coming steps we take the sign of the calculation but was included for completeness
-    else:
-      direction = 0 
+    """
+    Calculates the gradient of loss with respect to a singular scalar 
+    feature embedding. Used to calculate FGSM perturbation.
+
+    Args:
+      model_builder: The model in use
+      answer_num: Integer representing answer to be perturbed.
+      feature_num: Integer for feature for which to calculate gradient with respect to. 
+      Should be a number between 20-39, corresponding to embedding dimension.
+
+    Returns:
+      Gradient of loss with respect to singular feature embedding.
+    """
+    features = model_builder.normalized_features_evaluated
+    direction = 0
+    layer_size = constants._HIDDEN_LAYER_DIMS[0]
+    for num in range(layer_size):
+        grad = model_builder.grad_variable_pair_evaluated[0][feature_num][num]
+        weight = model_builder.grad_variable_pair_evaluated[1][feature_num][num]
+        feature = features[answer_num][feature_num]
+        if feature != 0:
+            # mathmatically feature value has no impact here since in coming
+            # steps we take the sign of the calculation but was included for
+            # completeness
+            direction += grad * (weight / feature)
+        else:
+            direction = 0
+
+    return direction
 
-  return direction
 
 def _get_perturb_dir(model_builder, answer_num):
-  """
-  Calculates gradient of loss with respect to features for entire answer. Used to calculate FGSM perturbation.
+    """
+    Calculates gradient of loss with respect to features for entire answer. 
+    Used to calculate FGSM perturbation.
+
+    Args:
+      model_builder: The model in use.
+      answer_num: Integer representing answer to be perturbed.
+
+    Returns:
+      Array of length 40 where the final 20 entries correspond to gradient of loss wrt to 
+      features that will be applied as noise to answer embeddings. First 20 entries 
+      correspond to noise applied to question embedding and thus are filled with 0s.
+    """
+    direction = []
+    for num in range(constants._EMBEDDING_DIMENSION):
+        # Corresponds to question embedding which we do not want to perturb

```

Comment:
Move this out of the for loop,  also let's make sure all comments have . at end.

Reply:
Same everywhere, let's try put comments outside if or for.